### PR TITLE
feat(internal): make datasource loading syncronous

### PR DESCRIPTION
**Pull request by viceice**
* Load `datasource` synchronous like `manager` and `versioning`.
* Unify dynamic code generation.

ref: #6747

### CODE REVIEW

**Review by zharinov — APPROVED** [review]
lgtm

File: lib/datasource/index.ts [hunk 1]
```diff
@@ -292,7 +291,7 @@ export async function getDigest(
   );
 }
 
-export async function getDefaultConfig(datasource: string): Promise<object> {
-  const loadedDatasource = await load(datasource);
-  return loadedDatasource?.defaultConfig || {};
+export function getDefaultConfig(datasource: string): Promise<object> {
+  const loadedDatasource = load(datasource);
+  return Promise.resolve(loadedDatasource?.defaultConfig || {});
```
Comment (viceice) [inline lib/datasource/index.ts:296]: Can technically be non promised function, but that will produce a lot of changes which i would like to do in a seperate pr.

**Comment by rarkins** [comment]
If it's invisible to the user and works, lgtm

**Comment by viceice** [comment]
sure, the functionallity hasn't changed.

**Comment by viceice** [comment]
i need an approve from @rarkins or @JamieMagee cause @zharinov's review doesn't count 😉

**Review by rarkins — APPROVED** [review]

**Pull request merged**